lib/call-cabal-project-to-nix.nix

```diff
@@ -527,23 +562,31 @@ let
     export SSL_CERT_FILE=${cacert}/etc/ssl/certs/ca-bundle.crt
     export GIT_SSL_CAINFO=${cacert}/etc/ssl/certs/ca-bundle.crt
 
-    # Using `cabal v2-freeze` will configure the project (since
-    # it is not configured yet), taking the existing `cabal.project.freeze`
-    # file into account.  Then it "writes out a freeze file which
-    # records all of the versions and flags that are picked" (from cabal docs).
+    export CABAL_DIR=$(mktemp -d)
+
+    cat >$CABAL_DIR/config <<EOF
+    repository hackage.haskell.org
+      url: http://hackage.haskell.org/
+      secure: True
+      root-keys: aaa
+      key-threshold: 0
+    EOF
+    cat $CABAL_DIR/config
+
+    echo "Installing fake curl"
+    PATH=${
+      fakeCurl ({
+        "http://hackage.haskell.org" = hackageRepo { index-state = cached-index-state; sha256 = index-sha256-found; };
+      } // inputMap)
+    }/bin:$PATH
+
+    cabal update \
+        -w ${ghc.targetPrefix}ghc \
+        --with-ghc-pkg=${ghc.targetPrefix}ghc-pkg
```

Comment:
cabal update needs the compile, mistery

Reply:
That is https://github.com/haskell/cabal/issues/8352. At least in this case we have a ghc we need in the derivation anyway.  Grep haskell.nix for `cabal-issue-8352-workaround` to find all the places we did not.

Reply:
Ah right, now I remember. I guess I'll add a comment and leave it like that. In principle not even make-install-plan should need a real compiler, it only needs a compiler version to resolve the conditionals in the package descriptions and the project files.